### ISSUE TITLE
Error in Matrix(from) : invalid 'data' in CreateSeuratObject

### ISSUE THREAD

**Issue by mtoseef99** [satijalab/seurat#6620]
I am new to single-cell analysis and Seurat and trying to create a Seurat object from GEODatasets, accession number GSE120575. I loaded the raw count matrix and then created a count matrix with the following code.

```r
# loading data with read.table 
sf_dat <- read.table("/Users/mtoseef/Data/GSE120575/GSE120575_Sade_Feldman_melanoma_single_cells_TPM_GEO.txt", fill = TRUE, header = F, sep = "\t")

# vectors of gens and cells
genes <- sf_dat[c(-1, -2), 1]
cells <- as.vector(t(sf_dat[1, 2:16292]))

# creating a count matrix
sf_dat.modified <- sf_dat[c(-1, -2), 2:16292]
colnames(sf_dat.modified) <- cells
rownames(sf_dat.modified) <- genes
```

```r
> dim(sf_dat.modified)
[1] 55737 16291
```
When I tried to create a Seurat object with CreateSeuratObject, it gives me an invalid data error.

```r
#Create Seurat objec
sf.seurat.obj <- CreateSeuratObject(counts = sf_dat.modified, project = "SadeFeldman", min.cells = 10)
```

```
Error in Matrix(from) : invalid 'data'
```
Any suggestions to solve this?

**Comment by mtoseef99** [comment]
I am sharing the traceback error message.

```r
Error in Matrix(from) : invalid 'data'
11. stop("invalid 'data'")
10. Matrix(from)
9. asMethod(object)
8. as(object = x, Class = "Matrix")
7. as.sparse.matrix(x = as.matrix(x = x))
6. as.sparse(x = as.matrix(x = x))
5. as.sparse.data.frame(x = counts, ...)
4. as.sparse(x = counts, ...)
3. CreateAssayObject(counts = counts, min.cells = min.cells, min.features = min.features, row.names = row.names)
2. CreateSeuratObject.default(counts = exp.mat, project = "SadeFeldman", min.cells = 10)
1. CreateSeuratObject(counts = exp.mat, project = "SadeFeldman", min.cells = 10)
```

**Comment by apposada** [comment]
Hi,

Our lab is experiencing this same issue. We have been able to replicate this in four independent installations of R. I have tried to load data using different versions of the Seurat package (down to 4.0.3) to no avail. I am unfamiliar with Seurat but, provided it happens with multiple versions of Seurat, I think it might be related to changes in some of the dependency packages.

**Comment by dasalam42** [comment]
Hi,

Expanding a bit on the previous comment of my lab partner @apposada, we have been troubleshooting this error downloading different version of Seurat. All these attempts were made in different machines, and threw the same error mentioned. 

We found a temporal solution without knowing what is causing it by manually copying a conda environment setup with previous versions of the dependencies listed below. All fresh installations of R and/or Seurat retrieve the same error. 

Many thanks 

```
abind | 1.4-5
AnnotationDbi | 1.54.1
ape | 5.5
aplot | 0.1.1
askpass | 1.1
assertthat | 0.2.1
backports | 1.2.1
base64enc | 0.1-3
beachmat | 2.8.1
BH | 1.75.0-0
Biobase | 2.52.0
BiocFileCache | 2.0.0
BiocGenerics | 0.38.0
BiocIO | 1.2.0
BiocManager | 1.30.16
BiocParallel | 1.26.2
BiocVersion | 3.13.1
biomaRt | 2.48.3
Biostrings | 2.60.2
bit | 4.0.4
bit64 | 4.0.5
bitops | 1.0-7
blob | 1.2.2
brew | 1.0-6
brio | 1.1.2
bslib | 0.3.0
cachem | 1.0.6
callr | 3.7.0
caTools | 1.18.2
checkmate | 2.0.0
chron | 2.3-56
cli | 3.1.0
clipr | 0.7.1
clusterProfiler | 4.0.5
clustree | 0.4.3
colorspace | 2.0-2
commonmark | 1.7
cowplot | 1.1.1
cpp11 | 0.4.0
crayon | 1.4.2
credentials | 1.3.1
crosstalk | 1.1.1
curl | 4.3.2
data.table | 1.14.2
DBI | 1.1.1
dbplyr | 2.1.1
DeconRNASeq | 1.34.0
DelayedArray | 0.18.0
DelayedMatrixStats | 1.14.3
deldir | 0.2-10
desc | 1.4.0
devtools | 2.4.2
diffobj | 0.3.5
digest | 0.6.28
DO.db | 2.9
doParallel | 1.0.16
DOSE | 3.18.3
dotCall64 | 1.0-1
DoubletDecon | 1.1.6
DoubletFinder | 2.0.3
downloader | 0.4
dplyr | 1.0.7
dqrng | 0.3.0
DropletUtils | 1.12.3
edgeR | 3.34.1
ellipsis | 0.3.2
enrichplot | 1.12.3
evaluate | 0.14
expm | 0.999-6
fansi | 0.5.0
farver | 2.1.0
fastmap | 1.1.0
fastmatch | 1.1-3
fgsea | 1.18.0
fields | 13.3
filelock | 1.0.2
fitdistrplus | 1.1-6
FNN | 1.1.3
fontawesome | 0.2.2
foreach | 1.5.1
formatR | 1.11
Formula | 1.2-4
fs | 1.5.0
futile.logger | 1.4.3
futile.options | 1.0.1
future | 1.22.1
future.apply | 1.8.1
generics | 0.1.0
GenomeInfoDb | 1.28.4
GenomeInfoDbData | 1.2.6
GenomicAlignments | 1.28.0
GenomicFeatures | 1.44.2
GenomicRanges | 1.44.0
gert | 1.4.1
ggforce | 0.3.3
ggfun | 0.0.4
ggplot2 | 3.3.5
ggplotify | 0.1.0
ggraph | 2.0.5
ggrepel | 0.9.1
ggridges | 0.5.3
ggtree | 3.0.4
gh | 1.3.0
gitcreds | 0.1.1
glmGamPoi | 1.4.0
globals | 0.14.0
glue | 1.4.2
GO.db | 3.13.0
goftest | 1.2-2
GOSemSim | 2.18.1
gplots | 3.1.1
graphlayouts | 0.7.1
gridExtra | 2.3
gridGraphics | 0.5-1
gsubfn | 0.7
gtable | 0.3.0
gtools | 3.9.2
HDF5Array | 1.20.0
here | 1.0.1
highr | 0.9
Hmisc | 4.6-0
hms | 1.1.1
htmlTable | 2.3.0
htmltools | 0.5.2
htmlwidgets | 1.5.4
httpuv | 1.6.3
httr | 1.4.2
ica | 1.0-2
igraph | 1.2.6
ini | 0.3.1
IRanges | 2.26.0
IRdisplay | 1
IRkernel | 1.2
irlba | 2.3.3
isoband | 0.2.5
iterators | 1.0.13
jpeg | 0.1-9
jquerylib | 0.1.4
jsonlite | 1.7.2
KEGGREST | 1.32.0
knitr | 1.36
labeling | 0.4.2
lambda.r | 1.2.4
later | 1.3.0
latticeExtra | 0.6-29
lazyeval | 0.2.2
leiden | 0.3.9
lifecycle | 1.0.1
limma | 3.48.3
limSolve | 1.5.6
listenv | 0.8.0
lmtest | 0.9-38
locfit | 1.5-9.4
lpSolve | 5.6.15
magrittr | 2.0.1
maps | 3.4.0
MatrixGenerics | 1.4.3
matrixStats | 0.61.0
MCL | 1
memoise | 2.0.0
mime | 0.12
miniUI | 0.1.1.1
munsell | 0.5.0
mygene | 1.28.0
openssl | 1.4.5
parallelly | 1.28.1
patchwork | 1.1.1
pbapply | 1.5-0
pbdZMQ | 0.3-5
pcaMethods | 1.84.0
pillar | 1.6.4
pkgbuild | 1.2.0
pkgconfig | 2.0.3
pkgload | 1.2.3
plogr | 0.2.0
plotly | 4.9.4.1
plyr | 1.8.6
png | 0.1-7
polyclip | 1.10-0
praise | 1.0.0
prettyunits | 1.1.1
processx | 3.5.2
progress | 1.2.2
promises | 1.2.0.1
proto | 1.0.0
ps | 1.6.0
purrr | 0.3.4
quadprog | 1.5-8
qvalue | 2.24.0
R.methodsS3 | 1.8.1
R.oo | 1.24.0
R.utils | 2.11.0
R6 | 2.5.1
RANN | 2.6.1
rappdirs | 0.3.3
rcmdcheck | 1.4.0
RColorBrewer | 1.1-2
Rcpp | 1.0.7
RcppAnnoy | 0.0.19
RcppArmadillo | 0.10.7.0.0
RcppEigen | 0.3.3.9.1
RcppProgress | 0.4.2
RCurl | 1.98-1.5
rematch2 | 2.1.2
remotes | 2.4.1
renv | 0.14.0
repr | 1.1.3
reshape2 | 1.4.4
restfulr | 0.0.13
reticulate | 1.22
rhdf5 | 2.36.0
rhdf5filters | 1.4.0
Rhdf5lib | 1.14.2
Rhtslib | 1.24.0
rjson | 0.2.20
rlang | 0.4.12
ROCR | 1.0-11
roxygen2 | 7.1.2
rprojroot | 2.0.2
Rsamtools | 2.8.0
RSpectra | 0.16-0
RSQLite | 2.2.8
rstudioapi | 0.13
rtracklayer | 1.52.1
Rtsne | 0.15
rversions | 2.1.1
S4Vectors | 0.30.2
sass | 0.4.0
scales | 1.1.1
scattermore | 0.7
scatterpie | 0.1.7
sctransform | 0.3.2
scuttle | 1.2.1
sessioninfo | 1.1.1
Seurat | 4.0.4
SeuratObject | 4.0.2
shadowtext | 0.0.9
shiny | 1.7.0
shinycssloaders | 1.0.0
shinyDirectoryInput | 0.2.0
shinyFiles | 0.9.0
shinyjs | 2.0.0
SingleCellExperiment | 1.14.1
sitmo | 2.0.1
snow | 0.4-3
sourcetools | 0.1.7
spam | 2.7-0
sparseMatrixStats | 1.4.2
spatstat.core | 2.3-0
spatstat.data | 2.1-0
spatstat.geom | 2.2-2
spatstat.sparse | 2.0-0
spatstat.utils | 2.2-0
sqldf | 0.4-11
stringi | 1.7.5
stringr | 1.4.0
SummarizedExperiment | 1.22.0
sys | 3.4
tensor | 1.5
testthat | 3.1.0
tibble | 3.1.5
tidygraph | 1.2.0
tidyr | 1.1.4
tidyselect | 1.1.1
tidytree | 0.3.5
treeio | 1.16.2
tweenr | 1.0.2
umap | 0.2.7.0
usethis | 2.0.1
utf8 | 1.2.2
uuid | 0.1-4
uwot | 0.1.10
V8 | 3.4.2
vctrs | 0.3.8
viridis | 0.6.2
viridisLite | 0.4.0
waldo | 0.3.1
whisker | 0.4
withr | 2.4.2
writexl | 1.4.0
xfun | 0.26
XML | 3.99-0.8
xml2 | 1.3.2
xopen | 1.0.0
xtable | 1.8-4
XVector | 0.32.0
yaml | 2.2.1
yulab.utils | 0.0.4
zip | 2.2.0
zlibbioc | 1.38.0
zoo | 1.8-9
```

**Comment by timoast** [comment]
>  We have been able to replicate this in four independent installations of R, using several independently-generated datasets (that used to work previously as-is) as well as a dummy matrix object.

@apposada if you're able to share the reproducible example using a dummy matrix that would be very helpful

**Comment by timoast** [comment]
I was able to reproduce this issue using the example from @mtoseef99, we will try to fix this soon

**Comment by timoast** [comment]
The problem is due to matrices containing character values, I've opened a pull request in the SeuratObject repo to fix it

**Comment by mtoseef99** [comment]
> The problem is due to matrices containing character values, I've opened a pull request in the SeuratObject repo to fix it

Thank you very much for looking into this but the problem still exists even after I pulled new `Seurat` and `SeuratObject` with `install.packages()`. Error is still the same:

```r
Error in Matrix(from) : invalid 'data'
```

Furthermore, I am providing the session info (RStudio).

```r
> sessionInfo()
R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_Hong Kong SAR.utf8  LC_CTYPE=English_Hong Kong SAR.utf8    LC_MONETARY=English_Hong Kong SAR.utf8
[4] LC_NUMERIC=C                           LC_TIME=English_Hong Kong SAR.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] gridExtra_2.3       sp_1.5-0            SeuratObject_4.1.2  Seurat_4.2.0        GEOquery_2.64.2     Biobase_2.56.0     
[7] BiocGenerics_0.44.0 ggplot2_3.3.6      

loaded via a namespace (and not attached):
  [1] Rtsne_0.16            colorspace_2.0-3      deldir_1.0-6          ellipsis_0.3.2        ggridges_0.5.4        rstudioapi_0.14      
  [7] spatstat.data_2.2-0   leiden_0.4.3          listenv_0.8.0         ggrepel_0.9.1         fansi_1.0.3           xml2_1.3.3           
 [13] codetools_0.2-18      splines_4.2.1         polyclip_1.10-0       jsonlite_1.8.2        ica_1.0-3             cluster_2.1.3        
 [19] png_0.1-7             rgeos_0.5-9           uwot_0.1.14           shiny_1.7.2           sctransform_0.3.5     spatstat.sparse_2.1-1
 [25] readr_2.1.3           compiler_4.2.1        httr_1.4.4            assertthat_0.2.1      Matrix_1.5-1          fastmap_1.1.0        
 [31] lazyeval_0.2.2        limma_3.52.4          cli_3.4.1             later_1.3.0           htmltools_0.5.3       tools_4.2.1          
 [37] igraph_1.3.5          gtable_0.3.1          glue_1.6.2            RANN_2.6.1            reshape2_1.4.4        dplyr_1.0.10         
 [43] Rcpp_1.0.9            scattermore_0.8       vctrs_0.4.2           nlme_3.1-157          progressr_0.11.0      lmtest_0.9-40        
 [49] spatstat.random_2.2-0 stringr_1.4.1         globals_0.16.1        mime_0.12             miniUI_0.1.1.1        lifecycle_1.0.3      
 [55] irlba_2.3.5.1         goftest_1.2-3         future_1.28.0         MASS_7.3-57           zoo_1.8-11            scales_1.2.1         
 [61] spatstat.core_2.4-4   hms_1.1.2             promises_1.2.0.1      spatstat.utils_2.3-1  parallel_4.2.1        RColorBrewer_1.1-3   
 [67] reticulate_1.26       pbapply_1.5-0         rpart_4.1.16          stringi_1.7.8         rlang_1.0.6           pkgconfig_2.0.3      
 [73] matrixStats_0.62.0    lattice_0.20-45       ROCR_1.0-11           purrr_0.3.5           tensor_1.5            patchwork_1.1.2      
 [79] htmlwidgets_1.5.4     cowplot_1.1.1         tidyselect_1.2.0      parallelly_1.32.1     RcppAnnoy_0.0.19      plyr_1.8.7           
 [85] magrittr_2.0.3        R6_2.5.1              generics_0.1.3        DBI_1.1.3             mgcv_1.8-40           pillar_1.8.1         
 [91] withr_2.5.0           fitdistrplus_1.1-8    survival_3.3-1        abind_1.4-5           tibble_3.1.8          future.apply_1.9.1   
 [97] KernSmooth_2.23-20    utf8_1.2.2            spatstat.geom_2.4-0   plotly_4.10.0         tzdb_0.3.0            grid_4.2.1           
[103] data.table_1.14.2     digest_0.6.29         xtable_1.8-4          tidyr_1.2.1           httpuv_1.6.6          munsell_0.5.0        
[109] viridisLite_0.4.1  
```

**Comment by mtoseef99** [comment]
> Hi,
> 
> Our lab is experiencing this same issue. We have been able to replicate this in four independent installations of R, using several independently-generated datasets (that used to work previously as-is) as well as a dummy matrix object. I have tried to load data using different versions of the Seurat package (down to 4.0.3) to no avail. I am unfamiliar with Seurat but, provided it happens with multiple versions of Seurat, I think it might be related to changes in some of the dependency packages.
> 
> EDIT just saw the other issue from Sep 23 #6420 . I want to clarify that in our case we only have numeric integers in the matrices. Also, just to be clear, we use seurat in a step of a pipeline that the lab runs routinely. I can provide the commands we use in that step if needed.
> 
> Thanks in advance!

Thank you for pointing out #6420 but I already checked that I have numeric values in the count matrix, which can be checked with the following.

```r
> sf_dat.modified[1:10, 1:10]
         A10_P3_M11 A11_P1_M11 A11_P3_M11 A11_P4_M11 A12_P3_M11 A12_P6_M11 A2_P1_M11 A2_P4_M11 A3_P1_M11 A3_P3_M11
TSPAN6         0.00       0.00       0.00       0.00       0.00       0.00      0.00      0.00      0.00      0.00
TNMD           0.00       0.00       0.00       0.00       0.00       0.00      0.00      0.00      0.00      0.00
DPM1           0.00       9.24       0.00       7.99       0.00       0.00      0.00      0.00      8.44      0.00
SCYL3          0.00       0.00       0.00       0.00       0.00       0.00      0.00      0.00      0.00      0.00
C1orf112       0.00       0.00       0.00       0.00       0.00       0.00      0.00      0.00      0.00      0.00
FGR            0.00       0.00       0.00       0.00       0.00       0.00      0.00      0.00      8.24      0.00
CFH            0.00       0.00       0.00       0.00       0.00       0.00      0.00      0.00      0.00      0.00
FUCA2          0.00       0.00       0.00       0.00       0.00       0.00      0.00      0.00      0.00      0.00
GCLC           0.00       1.01       0.00       0.00       0.00       0.00      0.00      0.00      0.00      0.00
NFYA           0.00       0.00       0.00       0.00       0.00       0.00      0.00      0.00      0.00      6.42
``` 

> EDIT just saw the other issue from Sep 23 #6420 . I want to clarify that in our case we only have numeric integers in the matrices. Also, just to be clear, we use seurat in a step of a pipeline that the lab runs routinely. I can provide the commands we use in that step if needed.

It would be much appreciated, could you please provide the seurat pipeline commands? Many thanks : )

**Comment by mtoseef99** [comment]
> Hi,
> 
> Expanding a bit on the previous comment of my lab partner @apposada, we have been troubleshooting this error downloading different version of Seurat. All these attempts were made in different machines, and threw the same error mentioned.
> 
> We found a temporal solution without knowing what is causing it by manually copying a conda environment setup with previous versions of the dependencies listed below. All fresh installations of R and/or Seurat retrieve the same error.
> 
> Many thanks
> 
> ```
> abind | 1.4-5
> AnnotationDbi | 1.54.1
> ape | 5.5
> aplot | 0.1.1
> askpass | 1.1
> assertthat | 0.2.1
> backports | 1.2.1
> base64enc | 0.1-3
> beachmat | 2.8.1
> BH | 1.75.0-0
> Biobase | 2.52.0
> BiocFileCache | 2.0.0
> BiocGenerics | 0.38.0
> BiocIO | 1.2.0
> BiocManager | 1.30.16
> BiocParallel | 1.26.2
> BiocVersion | 3.13.1
> biomaRt | 2.48.3
> Biostrings | 2.60.2
> bit | 4.0.4
> bit64 | 4.0.5
> bitops | 1.0-7
> blob | 1.2.2
> brew | 1.0-6
> brio | 1.1.2
> bslib | 0.3.0
> cachem | 1.0.6
> callr | 3.7.0
> caTools | 1.18.2
> checkmate | 2.0.0
> chron | 2.3-56
> cli | 3.1.0
> clipr | 0.7.1
> clusterProfiler | 4.0.5
> clustree | 0.4.3
> colorspace | 2.0-2
> commonmark | 1.7
> cowplot | 1.1.1
> cpp11 | 0.4.0
> crayon | 1.4.2
> credentials | 1.3.1
> crosstalk | 1.1.1
> curl | 4.3.2
> data.table | 1.14.2
> DBI | 1.1.1
> dbplyr | 2.1.1
> DeconRNASeq | 1.34.0
> DelayedArray | 0.18.0
> DelayedMatrixStats | 1.14.3
> deldir | 0.2-10
> desc | 1.4.0
> devtools | 2.4.2
> diffobj | 0.3.5
> digest | 0.6.28
> DO.db | 2.9
> doParallel | 1.0.16
> DOSE | 3.18.3
> dotCall64 | 1.0-1
> DoubletDecon | 1.1.6
> DoubletFinder | 2.0.3
> downloader | 0.4
> dplyr | 1.0.7
> dqrng | 0.3.0
> DropletUtils | 1.12.3
> edgeR | 3.34.1
> ellipsis | 0.3.2
> enrichplot | 1.12.3
> evaluate | 0.14
> expm | 0.999-6
> fansi | 0.5.0
> farver | 2.1.0
> fastmap | 1.1.0
> fastmatch | 1.1-3
> fgsea | 1.18.0
> fields | 13.3
> filelock | 1.0.2
> fitdistrplus | 1.1-6
> FNN | 1.1.3
> fontawesome | 0.2.2
> foreach | 1.5.1
> formatR | 1.11
> Formula | 1.2-4
> fs | 1.5.0
> futile.logger | 1.4.3
> futile.options | 1.0.1
> future | 1.22.1
> future.apply | 1.8.1
> generics | 0.1.0
> GenomeInfoDb | 1.28.4
> GenomeInfoDbData | 1.2.6
> GenomicAlignments | 1.28.0
> GenomicFeatures | 1.44.2
> GenomicRanges | 1.44.0
> gert | 1.4.1
> ggforce | 0.3.3
> ggfun | 0.0.4
> ggplot2 | 3.3.5
> ggplotify | 0.1.0
> ggraph | 2.0.5
> ggrepel | 0.9.1
> ggridges | 0.5.3
> ggtree | 3.0.4
> gh | 1.3.0
> gitcreds | 0.1.1
> glmGamPoi | 1.4.0
> globals | 0.14.0
> glue | 1.4.2
> GO.db | 3.13.0
> goftest | 1.2-2
> GOSemSim | 2.18.1
> gplots | 3.1.1
> graphlayouts | 0.7.1
> gridExtra | 2.3
> gridGraphics | 0.5-1
> gsubfn | 0.7
> gtable | 0.3.0
> gtools | 3.9.2
> HDF5Array | 1.20.0
> here | 1.0.1
> highr | 0.9
> Hmisc | 4.6-0
> hms | 1.1.1
> htmlTable | 2.3.0
> htmltools | 0.5.2
> htmlwidgets | 1.5.4
> httpuv | 1.6.3
> httr | 1.4.2
> ica | 1.0-2
> igraph | 1.2.6
> ini | 0.3.1
> IRanges | 2.26.0
> IRdisplay | 1
> IRkernel | 1.2
> irlba | 2.3.3
> isoband | 0.2.5
> iterators | 1.0.13
> jpeg | 0.1-9
> jquerylib | 0.1.4
> jsonlite | 1.7.2
> KEGGREST | 1.32.0
> knitr | 1.36
> labeling | 0.4.2
> lambda.r | 1.2.4
> later | 1.3.0
> latticeExtra | 0.6-29
> lazyeval | 0.2.2
> leiden | 0.3.9
> lifecycle | 1.0.1
> limma | 3.48.3
> limSolve | 1.5.6
> listenv | 0.8.0
> lmtest | 0.9-38
> locfit | 1.5-9.4
> lpSolve | 5.6.15
> magrittr | 2.0.1
> maps | 3.4.0
> MatrixGenerics | 1.4.3
> matrixStats | 0.61.0
> MCL | 1
> memoise | 2.0.0
> mime | 0.12
> miniUI | 0.1.1.1
> munsell | 0.5.0
> mygene | 1.28.0
> openssl | 1.4.5
> parallelly | 1.28.1
> patchwork | 1.1.1
> pbapply | 1.5-0
> pbdZMQ | 0.3-5
> pcaMethods | 1.84.0
> pillar | 1.6.4
> pkgbuild | 1.2.0
> pkgconfig | 2.0.3
> pkgload | 1.2.3
> plogr | 0.2.0
> plotly | 4.9.4.1
> plyr | 1.8.6
> png | 0.1-7
> polyclip | 1.10-0
> praise | 1.0.0
> prettyunits | 1.1.1
> processx | 3.5.2
> progress | 1.2.2
> promises | 1.2.0.1
> proto | 1.0.0
> ps | 1.6.0
> purrr | 0.3.4
> quadprog | 1.5-8
> qvalue | 2.24.0
> R.methodsS3 | 1.8.1
> R.oo | 1.24.0
> R.utils | 2.11.0
> R6 | 2.5.1
> RANN | 2.6.1
> rappdirs | 0.3.3
> rcmdcheck | 1.4.0
> RColorBrewer | 1.1-2
> Rcpp | 1.0.7
> RcppAnnoy | 0.0.19
> RcppArmadillo | 0.10.7.0.0
> RcppEigen | 0.3.3.9.1
> RcppProgress | 0.4.2
> RCurl | 1.98-1.5
> rematch2 | 2.1.2
> remotes | 2.4.1
> renv | 0.14.0
> repr | 1.1.3
> reshape2 | 1.4.4
> restfulr | 0.0.13
> reticulate | 1.22
> rhdf5 | 2.36.0
> rhdf5filters | 1.4.0
> Rhdf5lib | 1.14.2
> Rhtslib | 1.24.0
> rjson | 0.2.20
> rlang | 0.4.12
> ROCR | 1.0-11
> roxygen2 | 7.1.2
> rprojroot | 2.0.2
> Rsamtools | 2.8.0
> RSpectra | 0.16-0
> RSQLite | 2.2.8
> rstudioapi | 0.13
> rtracklayer | 1.52.1
> Rtsne | 0.15
> rversions | 2.1.1
> S4Vectors | 0.30.2
> sass | 0.4.0
> scales | 1.1.1
> scattermore | 0.7
> scatterpie | 0.1.7
> sctransform | 0.3.2
> scuttle | 1.2.1
> sessioninfo | 1.1.1
> Seurat | 4.0.4
> SeuratObject | 4.0.2
> shadowtext | 0.0.9
> shiny | 1.7.0
> shinycssloaders | 1.0.0
> shinyDirectoryInput | 0.2.0
> shinyFiles | 0.9.0
> shinyjs | 2.0.0
> SingleCellExperiment | 1.14.1
> sitmo | 2.0.1
> snow | 0.4-3
> sourcetools | 0.1.7
> spam | 2.7-0
> sparseMatrixStats | 1.4.2
> spatstat.core | 2.3-0
> spatstat.data | 2.1-0
> spatstat.geom | 2.2-2
> spatstat.sparse | 2.0-0
> spatstat.utils | 2.2-0
> sqldf | 0.4-11
> stringi | 1.7.5
> stringr | 1.4.0
> SummarizedExperiment | 1.22.0
> sys | 3.4
> tensor | 1.5
> testthat | 3.1.0
> tibble | 3.1.5
> tidygraph | 1.2.0
> tidyr | 1.1.4
> tidyselect | 1.1.1
> tidytree | 0.3.5
> treeio | 1.16.2
> tweenr | 1.0.2
> umap | 0.2.7.0
> usethis | 2.0.1
> utf8 | 1.2.2
> uuid | 0.1-4
> uwot | 0.1.10
> V8 | 3.4.2
> vctrs | 0.3.8
> viridis | 0.6.2
> viridisLite | 0.4.0
> waldo | 0.3.1
> whisker | 0.4
> withr | 2.4.2
> writexl | 1.4.0
> xfun | 0.26
> XML | 3.99-0.8
> xml2 | 1.3.2
> xopen | 1.0.0
> xtable | 1.8-4
> XVector | 0.32.0
> yaml | 2.2.1
> yulab.utils | 0.0.4
> zip | 2.2.0
> zlibbioc | 1.38.0
> zoo | 1.8-9
> ```

@dasalam42 could you please provide the working example of the same dataset? And you run it with RStudio or conda environment?

Many thanks.

**Comment by mtoseef99** [comment]
> Hi,
> 
> Our lab is experiencing this same issue. We have been able to replicate this in four independent installations of R, using several independently-generated datasets (that used to work previously as-is) as well as a dummy matrix object. I have tried to load data using different versions of the Seurat package (down to 4.0.3) to no avail. I am unfamiliar with Seurat but, provided it happens with multiple versions of Seurat, I think it might be related to changes in some of the dependency packages.
> 

Thank you for pointing #6420, yes I can see the `"character"`, but when I tried converting it into to sparse matrix if gives me the same error (shown below). Do you have any idea why this dataset has responded like this?

```r
> sf.dat.sparse <- as.sparse(sf.dat.modified)
Error in Matrix(from) : invalid 'data'
```

> EDIT just saw the other issue from Sep 23 #6420 . I want to clarify that in our case we only have numeric integers in the matrices. Also, just to be clear, we use seurat in a step of a pipeline that the lab runs routinely. I can provide the commands we use in that step if needed.
> 
> Thanks in advance!

It would be much appreciated, could you please provide the Seurat pipeline commands? Thank you : )

**Comment by timoast** [comment]
@mtoseef99 it looks like you still have the old version loaded, you need to install from the develop branch of SeuratObject. The version with the fix is 4.1.2.9005

**Comment by mtoseef99** [comment]
> @mtoseef99 it looks like you still have the old version loaded, you need to install from the develop branch of SeuratObject. The version with the fix is 4.1.2.9005

@timoast thank you so much, yes after updating SeuratObject from the develop branch, I have successfully created seurat object.

**NOTE:** To download seurat-object, please follow as given below (if you haven't installed `remotes`)

```r
install.packages('remotes')
remotes::install_github(repo = 'mojaveazure/seurat-object', ref = 'develop')
```
